### ISSUE TITLE
Opening a theme file locks the initiating Leo session

### ISSUE THREAD

**Issue by ranvik14** [leo-editor/leo-editor#1425]
_Settings > Open A Theme File_ can look like it's not working.

That is I'm able to select a theme file, e.g. BreezeDark2Theme.leo - but - this outline is not opened within the Leo session that you're acting in. This command opens the theme .leo file outside of Leo in a separate process using:

    python launchLeo.py {fn}

This is done on purpose, so that the theme machinery can be tested in a clean environment. However the Leo session that spawns the Leo-with-themefile locks and says "Not Responding" in the title bar until the Leo-with-themefile session is closed.

**Comment by technatica** [comment]
When I open a theme file using  Settings > Open A Theme File, it opens the theme file just fine in a new instance of Leo.

Leo 6.1-b1-devel, devel branch, build b6a76cb9b2
2019-10-27 06:15:34 -0500
Python 3.6.8, PyQt version 5.13.1
linux

**Comment by ranvik14** [comment]
@technatica : Thanks for your reminder. - Here's the information for my system:

> Leo Log Window
> Leo 6.1-b1
> Python 3.7.5, PyQt version 5.13.1
> Windows 10 AMD64 (build 10.0.18362) SP0
> current dir: C:/Users/Viktor/pyve/Leo-6.1-b1
> load dir: C:/users/viktor/pyve/leo-6.1-b1/lib/site-packages/leo/core
> global config dir: C:/users/viktor/pyve/leo-6.1-b1/lib/site-packages/leo/config
> home dir: C:/Users/Viktor
> reading settings in C:/users/viktor/pyve/leo-6.1-b1/lib/site-packages/leo/config/leoSettings.leo
> reading settings in C:/Users/Viktor/.leo/myLeoSettings.leo
> reading settings in C:/Users/Viktor/Worklogs/WL2019.leo
> read outline in 5.06 seconds

In this environment the reported problem does exist !

**Comment by LewisNeal** [comment]
Before: Python 3.7.4, PyQt version 5.13.1 -  theme files were opening.
After:   Python 3.8.0, PyQt version 5.13.2 - theme files do not open.

Leo Log Window
Leo 6.1-b1-devel, devel branch, build dbc3685e3b
Windows 10 AMD64 (build 10.0.18362) SP0

**Comment by ranvik14** [comment]
@edreamleo : Here are some new findings for this problem, when using the latest"Leo-devel" archive.

1) Activate & prepare an existing PyVE to install the latest "Leo-devel" archive - OK. - See "Log-001"
2) Download, expand & install the version of "Leo-devel" - OK. - See "Log-002" & "Log-003"
3) Try to open a Leo theme file/outline - OK?

    * Kind of OK/ Not really OK.
    * Leo theme file/outline is created in a separate window - and not - as a separate tab of the existing window
    * Leo in existing window is unresponsive until you close the newly opened Leo theme file/outline ...

Please let me know, if you are interested in any other information from this PyVE.

**Comment by edreamleo** [comment]
@ranvik14 Thanks for this new data.  I'd like to delegate this to Matt.

**Comment by maphew** [comment]
It's going to take me awhile to clear the decks so I can dig in to this. 

In the meantime, it might help to stay with python 3.6 or 3.7. With a different project in a new conda env I learned pywin32 doesn't install properly on python 3.8. Telling conda to keep to 3.7 worked though, `conda install python=3.7`. [x-ref](https://github.com/adamrehn/ue4-docker/issues/53)).

**Comment by ranvik14** [comment]
@maphew : FYI. - This problem does occur with Python 3.7 in a Windows 10 environment as well as in a Windows 7 environment. - I don't have Python 3.8 installed on any of my systems yet.

**Comment by maphew** [comment]
Leo _"Settings >> Open a Them file  >> {select a file}"_ works for me. However I note the new session opens in a background window, behind all the other windows I have open including the current Leo session. This is easily missed when Taskbar is set to group similar windows together, is this perhaps what's happening for you @ranvik14 ?

~~~
Leo 6.2-b1-devel, devel branch, build 1667e84609
2019-12-18 17:54:47 -0600
Python 3.6.2, PyQt version 5.12.3
Windows 10 AMD64 (build 10.0.17134) SP0
~~~

**Comment by maphew** [comment]
Ahhh, I didn't read closely enough. In https://github.com/leo-editor/leo-editor/issues/1425#issuecomment-552191705 you're aware of this. The issue in that comment is slightly different from the opening post, namely: _opening a theme file locks the initating Leo session_. I'll update the title and OP.

**Comment by maphew** [comment]
As currently written this menu command, defined in `../commands/commanderFileCommands.py-->Themes-->c_file.open_theme_file`, will only work in Leo source code checkouts. That is, it's broken for anyone who installs from Pypi.org, or from `pip` in a non-editable mode, due to the assumption that `launchLeo.py` is present above ../leo, which is not possible in a PYTHONHOME env.

Exploration of fixes for this will carry on in https://groups.google.com/d/topic/leo-editor/E3xtlclrcSY/discussion.

**Comment by maphew** [comment]
Believe fixed with 016867f. Please test.

**Comment by edreamleo** [comment]
On Sun, Dec 22, 2019 at 12:47 AM matt wilkie <notifications@github.com>
wrote:

> Believe fixed with 016867f
> <https://github.com/leo-editor/leo-editor/commit/016867f5ade5da6796397ab4705287bf8b8f336a>.
> Please test.
>
Works for me.  Many thanks for this great work.

Edward

**Comment by ranvik14** [comment]
Hello Matt,

Am So., 22. Dez. 2019 um 06:47 Uhr schrieb matt wilkie <
notifications@github.com>:

> Believe fixed with 016867f
> <https://github.com/leo-editor/leo-editor/commit/016867f5ade5da6796397ab4705287bf8b8f336a>.
> Please test.
>
I downloaded the latest development archive from GitHub - and - installed
it via pip into a dedicated PyVE. - The reported problem when opening a
theme file is gone :-)

Thanks a lot from my side as well for resolving this issue!

With kind regards,

Viktor

**Comment by maphew** [comment]
Yay! It works for someone besides me. :) Thanks for testing.
